### PR TITLE
Add omero-py python package

### PR DESCRIPTION
**Pull request by joshmoore**
cc: @conda-forge/help-python (Resurrecting #13028)

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] ~If static libraries are linked in, the license of the static library is packaged.~
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.

## Background

This package was originally in bioconda, but we're moving it to conda-forge to support other packages, and also for Windows support: bioconda/bioconda-recipes#15831 (comment)

The ome has been maintaining its own repository for this package due to problems with the bioconda package:

 * https://github.com/ome/conda-omero-py
 * https://anaconda.org/ome/omero-py

But now that the dependency zeroc-ice 3.6 is available in conda-forge we'd like to add omero-py here too.

### CODE REVIEW

**Comment by joshmoore** [comment]
Looks like this is going to need a good deal of pinning in order to build:

```
Package ca-certificates conflicts for:
python[version='>=3.9,<3.10.0a0'] -> openssl[version='>=1.1.1j,<1.1.2a'] -> ca-certificates
pillow -> python[version='>=2.7,<2.8.0a0'] -> ca-certificates
pyyaml -> python[version='>=2.7,<2.8.0a0'] -> ca-certificates
future -> python[version='>=2.7,<2.8.0a0'] -> ca-certificates
requests -> python -> ca-certificates
appdirs -> python -> ca-certificates
numpy -> python[version='>=2.7,<2.8.0a0'] -> ca-certificates
zeroc-ice=3.6 -> openssl[version='>=1.1.1h,<1.1.2a'] -> ca-certificates

etc. etc. etc.
```

cc: @sbesson @jburel

**Comment by manics** [comment]
I think it's trying to build on pypy which won't work since there's no corresponding zero-ice package.

The `skip: true  # [py<36 or py>38 or python_impl=='pypy']` line should skip it but I suspect it's being ignored.

**Comment by joshmoore** [comment]
<details><summary>New failure:</summary>

```
Output in format: Requested package -> Available versions

Package python conflicts for:
pillow -> olefile -> python[version='3.7.*|3.9.*|3.8.*']
requests -> certifi[version='>=2017.4.17'] -> python[version='3.7.*|3.8.*|<4.0|>=3.9,<3.10.0a0']
numpy -> python[version='2.7.*|3.5.*|3.6.*|>=2.7,<2.8.0a0|>=3.6,<3.7.0a0|>=3.7,<3.8.0a0|>=3.9,<3.10.0a0|>=3.8,<3.9.0a0|>=3.5,<3.6.0a0|3.4.*']
requests -> python[version='2.7.*|3.5.*|3.6.*|>=2.7,<2.8.0a0|>=3.6,<3.7.0a0|>=3.8,<3.9.0a0|>=3.7,<3.8.0a0|>=3.5,<3.6.0a0|3.4.*']
numpy -> python_abi=3.9[build=*_cp39] -> python[version='3.7.*|3.8.*|3.9.*']
pillow -> python[version='2.7.*|3.5.*|3.6.*|>=2.7,<2.8.0a0|>=3.6,<3.7.0a0|>=3.7,<3.8.0a0|>=3.9,<3.10.0a0|>=3.8,<3.9.0a0|>=3.5,<3.6.0a0|3.4.*']
zeroc-ice=3.6 -> python_abi=3.7[build=*_cp37m] -> python[version='3.6.*|3.7.*|3.8.*']
pyyaml -> python_abi=3.7[build=*_cp37m] -> python[version='3.7.*|3.9.*|3.8.*']
future -> python_abi=3.9[build=*_cp39] -> python[version='3.7.*|3.9.*|3.8.*']
python[version='>=3.9,<3.10.0a0']
pyyaml -> python[version='2.7.*|3.5.*|3.6.*|>=2.7,<2.8.0a0|>=3.6,<3.7.0a0|>=3.7,<3.8.0a0|>=3.9,<3.10.0a0|>=3.8,<3.9.0a0|>=3.5,<3.6.0a0|3.4.*']
appdirs -> python[version='2.7.*|3.5.*|3.6.*|3.4.*|>=3.5,<3.6.0a0|>=2.7,<2.8.0a0|>=3.7,<3.8.0a0|>=3.6,<3.7.0a0']
zeroc-ice=3.6 -> python[version='>=2.7,<2.8.0a0|>=3.6,<3.7.0a0|>=3.7,<3.8.0a0|>=3.8,<3.9.0a0']
future -> python[version='2.7.*|3.4.*|3.5.*|3.6.*|>=2.7,<2.8.0a0|>=3.6,<3.7.0a0|>=3.7,<3.8.0a0|>=3.9,<3.10.0a0|>=3.8,<3.9.0a0|>=3.5,<3.6.0a0']

Package ca-certificates conflicts for:
zeroc-ice=3.6 -> openssl[version='>=1.1.1h,<1.1.2a'] -> ca-certificates
pyyaml -> python[version='>=2.7,<2.8.0a0'] -> ca-certificates
requests -> python -> ca-certificates
numpy -> python[version='>=2.7,<2.8.0a0'] -> ca-certificates
pillow -> python[version='>=2.7,<2.8.0a0'] -> ca-certificates
future -> python[version='>=2.7,<2.8.0a0'] -> ca-certificates
python[version='>=3.9,<3.10.0a0'] -> openssl[version='>=1.1.1j,<1.1.2a'] -> ca-certificates
appdirs -> python -> ca-certificates

...
```
</details>

File: recipes/omero-py/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    - VERSION_SUFFIX
+  entry_points:
+    - omero = omero.main:main
+  # zeroc-ice 3.6.5 is currently available for cpython 3.6, 3.7, 3.8
+  skip: True  # [py<36 or py>38 or python_impl=="pypy"]
+
+requirements:
+  host:
+    - pip
+    - python =3
```
Comment (dopplershift) [inline recipes/omero-py/meta.yaml:28]: Don't use the  `=3` here and below, we just rely on the `skip: True` above.

**Comment by manics** [comment]
There's a new attempt at this in https://github.com/conda-forge/staged-recipes/pull/17721

**Comment by joshmoore** [comment]
Closing in favor of #17721.

**Pull request closed without merging**